src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php

```diff
@@ -53,7 +53,7 @@ private function findAndSortTaggedServices($tagName, ContainerBuilder $container
         $services = [];
 
         foreach ($container->findTaggedServiceIds($tagName, true) as $serviceId => $attributes) {
-            $priority = isset($attributes[0]['priority']) ? $attributes[0]['priority'] : 0;
+            $priority = isset($attributes[0]['priority']) ? $attributes[0]['priority'] : (method_exists($serviceId, 'getDefaultPriority') ? $serviceId::getDefaultPriority() : 0);
```

Comment:
* The service id is not necessarily the FQCN; you should use the class of its definition instead (e.g. `$container->getDefinition($serviceId)->getClass()`).
* Looking for an ambitrary method can cause unexpected behaviors (i.e., what if my service uses this method for another purpose?) An alternative may be to add a new interface that forces this contract and look for this interface instead.

Reply:
The class is already extracted a few lines below, this should be reused.
Not sure about the interface Yonel, that wouldn't make much sense from the class' design pov I think.
Instead, I'd suggest supporting this only when a TaggedIteratorArgument is used, and add a getDefaultPriorityMethod on it. The default implementation would return `getDefaultMyTagPriority`, exactly as `getDefaultIndexMethod` returns `getDefaultMyTagName`

Reply:
I used `getClass` on my own implementation without using this trait and yes we could use `$class` extracted.
I wouldn't like the new interface, maybe a good solution is do the same of `getDefailtIndexMethod` like Nicolas says.

Reply:
@nicolas-grekas , am I right that what you suggest will be an expansion of https://github.com/symfony/symfony/pull/30257?

Reply:
Absolutely